pkg/syncer/spec/spec_process.go

```diff
@@ -298,6 +304,13 @@ func (c *Controller) ensureDownstreamNamespaceExists(ctx context.Context, downst
 		return fmt.Errorf("(namespace collision) namespace %s already exists, but has a different namespace locator annotation: %+v vs %+v", newNamespace.GetName(), nsLocator, desiredNSLocator)
 	}
 
+	// Handle kcp upgrades by checking the tenant ID is set and correct
+	if tenantID, ok := unstrNamespace.GetLabels()[shared.TenantIDLabel]; !ok || tenantID != desiredTenantID {
```

Comment:
Would there be a simple to add some unit test case that will check that this label is added to the downstream namespace if not already there ?

Reply:
done